python/tank_vendor/shotgun_authentication/shotgun_authenticator.py

```diff
@@ -129,9 +130,13 @@ def create_session_user(self, login, session_token=None, password=None, host=Non
         http_proxy = http_proxy or self._defaults_manager.get_http_proxy()
 
         # If we only have a password, generate a session token.
-        if password and not session_token:
+        if password is None and session_token is None:
+            # todo - find this in our 'phonebook' of stored login/session ids
+            raise AuthenticationError("No session details for this user.")
```

Comment:
Same here - there is a whole bunch of work to do on the security API side - but I just added the minimum implementation and left the rest to you since I know you are already on the case! Hope that makes sense!